mackerel-plugin-solr/lib/solr_test.go

```diff
@@ -0,0 +1,198 @@
+	default:
+		fmt.Fprintf(w, "")
+	}
+})
+
+func fetchJSON(name string) string {
+	path := fmt.Sprintf("./stats/%s/%s.json", solrVersion, name)
+	json, err := ioutil.ReadFile(path)
+	if err != nil {
+		panic(err)
+	}
+	return string(json)
+}
+
+func fetchJSONForMbeans(key string) string {
+	switch key {
+	case "QUERYHANDLER":
+		return fetchJSON("query")
+	case "UPDATEHANDLER":
+		return fetchJSON("update")
+	case "REPLICATION":
+		return fetchJSON("replication")
+	case "CACHE":
+		return fetchJSON("cache")
+	default:
+		return ""
+	}
+}
+
+func TestGraphDefinition(t *testing.T) {
+	solr := SolrPlugin{
+		Cores:  []string{"testcore"},
+		Prefix: "solr",
+	}
+	graphdef := solr.GraphDefinition()
+
+	assert.EqualValues(t, "testcore DocsCount", graphdef["solr.testcore.docsCount"].Label)
+	assert.EqualValues(t, "testcore_numDocs", graphdef["solr.testcore.docsCount"].Metrics[0].Name)
+	assert.EqualValues(t, "testcore_deletedDocs", graphdef["solr.testcore.docsCount"].Metrics[1].Name)
+
+	assert.EqualValues(t, "testcore IndexHeapUsageBytes", graphdef["solr.testcore.indexHeapUsageBytes"].Label)
+	assert.EqualValues(t, "testcore_indexHeapUsageBytes", graphdef["solr.testcore.indexHeapUsageBytes"].Metrics[0].Name)
+	assert.EqualValues(t, "testcore SegmentCount", graphdef["solr.testcore.segmentCount"].Label)
+	assert.EqualValues(t, "testcore_segmentCount", graphdef["solr.testcore.segmentCount"].Metrics[0].Name)
+	assert.EqualValues(t, "testcore SizeInBytes", graphdef["solr.testcore.sizeInBytes"].Label)
+	assert.EqualValues(t, "testcore_sizeInBytes", graphdef["solr.testcore.sizeInBytes"].Metrics[0].Name)
+
+	assert.EqualValues(t, "testcore Requests", graphdef["solr.testcore.requests"].Label)
+	assert.EqualValues(t, "testcore_requests_updatejson", graphdef["solr.testcore.requests"].Metrics[0].Name)
+	assert.EqualValues(t, "testcore_requests_select", graphdef["solr.testcore.requests"].Metrics[1].Name)
+	assert.EqualValues(t, "testcore_requests_updatejsondocs", graphdef["solr.testcore.requests"].Metrics[2].Name)
+	assert.EqualValues(t, "testcore_requests_get", graphdef["solr.testcore.requests"].Metrics[3].Name)
+	assert.EqualValues(t, "testcore_requests_updatecsv", graphdef["solr.testcore.requests"].Metrics[4].Name)
+	assert.EqualValues(t, "testcore_requests_replication", graphdef["solr.testcore.requests"].Metrics[5].Name)
+	assert.EqualValues(t, "testcore_requests_update", graphdef["solr.testcore.requests"].Metrics[6].Name)
+	assert.EqualValues(t, "testcore_requests_dataimport", graphdef["solr.testcore.requests"].Metrics[7].Name)
+
+	assert.EqualValues(t, "testcore Errors", graphdef["solr.testcore.errors"].Label)
+	assert.EqualValues(t, "testcore Timeouts", graphdef["solr.testcore.timeouts"].Label)
+	assert.EqualValues(t, "testcore AvgRequestsPerSecond", graphdef["solr.testcore.avgRequestsPerSecond"].Label)
+	assert.EqualValues(t, "testcore 5minRateRequestsPerSecond", graphdef["solr.testcore.5minRateRequestsPerSecond"].Label)
```

Comment:
http://lucene.apache.org/solr/community.html#about-versions

We can exclude version `5.*` in near the future. :smile: